cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/TopicReplicationFactorAnomaly.java

```diff
@@ -31,11 +31,14 @@
  */
 public class TopicReplicationFactorAnomaly extends TopicAnomaly {
   protected Short _targetReplicationFactor;
-  protected Set<String> _topicsWithBadReplicationFactor;
+  protected Map<Boolean, Set<String>> _badTopicsByFixability;
   protected UpdateTopicConfigurationRunnable _updateTopicConfigurationRunnable;
 
   @Override
   public boolean fix() throws Exception {
+    if (_updateTopicConfigurationRunnable == null) {
+      return false;
+    }
```

Comment:
When is this expected to happen? Should we instead throw illegal state exception here?

Reply:
This will happen if all the topic are unfixable, i.e. target RF is smaller than minISR